sktime/classification/early_classification/_teaser.py

```diff
@@ -0,0 +1,564 @@
+            threads = self._threads_to_use if m is None else 1
+
+            out = Parallel(n_jobs=threads)(
+                delayed(self._predict_proba_for_estimator)(
+                    X,
+                    i,
+                )
+                for i in range(idx + 1)
+            )
+
+            X_oc, probas, preds = zip(*out)
+
+            new_state_info, decisions = self._predict_n_timestamps(
+                preds, X_oc, self._consecutive_predictions, idx + 1
+            )
+
+            probas = np.array(
+                [probas[new_state_info[i][0]][i] for i in range(n_instances)]
+            )
+        else:
+            # if this is the smallest dataset, there should be no state_info, else we
+            # should have state info for each, and they should all be the same length
+            if idx == 0 and (state_info is None or state_info == []):
+                state_info = np.zeros((n_instances, 3), dtype=int)
+            elif isinstance(state_info, np.ndarray) and idx > 0:
+                if not all(si[0] == idx - 1 for si in state_info):
+                    raise ValueError(
+                        "All state_info input instances must be from the "
+                        "previous classification point series length."
+                    )
+            else:
+                raise ValueError(
+                    "state_info should be None or an empty List for first time input, "
+                    "and a list of state_info outputs from the previous decision "
+                    "making for later inputs."
+                )
+
+            probas = self._estimators[idx].predict_proba(X)
+
+            # find predicted class for each instance
+            rng = check_random_state(self.random_state)
+            preds = [
+                int(rng.choice(np.flatnonzero(prob == prob.max()))) for prob in probas
+            ]
+
+            # make a decision based on the one class classifier prediction
+            if self._one_class_classifiers[idx] is not None:
+                X_oc = self._generate_one_class_features(X, preds, probas)
+                decisions = np.array(
+                    self._one_class_classifiers[idx].predict(X_oc) == 1, dtype=bool
+                )
+            else:
+                decisions = np.zeros(n_instances, dtype=bool)
+
+            # record consecutive class decisions
+            new_state_info = np.array(
+                [
+                    self._update_state_info(decisions, preds, state_info, i, idx)
+                    for i in range(n_instances)
+                ]
+            )
+
+            # if we have the full series, always decide True
+            if idx == len(self._classification_points) - 1:
+                decisions = np.ones(n_instances, dtype=bool)
+            else:
+                decisions = np.array(
+                    [
+                        True
+                        if state_info[i][1] >= self._consecutive_predictions
```

Comment:
I am currently refactoring the code. And this is likely wrong? this should be new_state_info[i][1]?

Reply:
yes

Reply:
Thank you :)